src/test/java/com/flowingcode/vaadin/addons/gridexporter/test/GridExporterTest.java

```diff
@@ -0,0 +1,147 @@
+package com.flowingcode.vaadin.addons.gridexporter.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.flowingcode.vaadin.addons.gridexporter.GridExporter;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.dom.Element;
+import java.lang.reflect.Field;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("GridExporter Internal Tooltip State Tests")
+class GridExporterTest {
+
+    private Grid<Object> mockGrid;
+    private GridExporter<Object> exporter;
+
+    @SuppressWarnings("unchecked")
+    private static <T> T getPrivateField(Object obj, String fieldName) {
```

Comment:
Maybe we can use a helper class `GridExporterForTest extends  GridExporter` that provides the required getters through reflection, which would improve readability of the tests themselve. (This is just a suggestion, I'm happy to commit as is since the public API has not been polluted)